src/client/driver/command-executors/execute-action-command.js

```diff
@@ -2,118 +2,134 @@ import hammerhead from '../deps/hammerhead';
+
 import COMMAND_TYPE from '../../../test-run/commands/type';
 
-var Promise            = hammerhead.Promise;
-var XhrBarrier         = testCafeCore.XhrBarrier;
-var pageUnloadBarrier  = testCafeCore.pageUnloadBarrier;
-var positionUtils      = testCafeCore.positionUtils;
-var waitFor            = testCafeCore.waitFor;
-var ClickAutomation    = testCafeRunner.get('./automation/playback/click');
-var RClickAutomation   = testCafeRunner.get('./automation/playback/rclick');
-var DblClickAutomation = testCafeRunner.get('./automation/playback/dblclick');
-var HoverAutomation    = testCafeRunner.get('./automation/playback/hover');
-var ProgressPanel      = testCafeUI.ProgressPanel;
+var Promise                 = hammerhead.Promise;
+var nativeMethods           = hammerhead.nativeMethods;
+var XhrBarrier              = testCafeCore.XhrBarrier;
+var pageUnloadBarrier       = testCafeCore.pageUnloadBarrier;
+var positionUtils           = testCafeCore.positionUtils;
+var waitFor                 = testCafeCore.waitFor;
+var ClickAutomation         = testCafeRunner.get('./automation/playback/click');
+var RClickAutomation        = testCafeRunner.get('./automation/playback/rclick');
+var DblClickAutomation      = testCafeRunner.get('./automation/playback/dblclick');
+var DragToOffsetAutomation  = testCafeRunner.get('./automation/playback/drag/to-offset');
+var DragToElementAutomation = testCafeRunner.get('./automation/playback/drag/to-element');
+var HoverAutomation         = testCafeRunner.get('./automation/playback/hover');
+var ProgressPanel           = testCafeUI.ProgressPanel;
 
 
 const PROGRESS_PANEL_TEXT   = 'Waiting for the target element of the next action to appear';
 const CHECK_ELEMENT_DELAY   = 200;
 const CHECK_ELEMENT_TIMEOUT = 10000;
 
 
-function ensureElementExists (selector) {
+function ensureElementExists (selector, ErrorCtor) {
     return waitFor(selector, CHECK_ELEMENT_DELAY, CHECK_ELEMENT_TIMEOUT)
         .catch(() => {
-            throw new ActionElementNotFoundError();
+            throw new ErrorCtor();
         });
 }
 
-function ensureElementVisible (element, timeout) {
+function ensureElementVisible (element, timeout, ErrorCtor) {
     return waitFor(() => positionUtils.isElementVisible(element) ? element : null, CHECK_ELEMENT_DELAY, timeout)
         .catch(() => {
-            throw new ActionElementIsInvisibleError();
+            throw new ErrorCtor();
         });
 }
 
-function ensureElement (selector) {
-    var startTime     = new Date();
+function ensureElement (selector, NotFoundErrorCtor, IsInvisibleErrorCtor) {
+    var startTime = new Date();
+
+    return ensureElementExists(() => nativeMethods.eval.call(window, selector), NotFoundErrorCtor)
+        .then(element => {
+            var checkVisibilityTimeout = CHECK_ELEMENT_TIMEOUT - (new Date() - startTime);
+
+            return ensureElementVisible(element, checkVisibilityTimeout, IsInvisibleErrorCtor);
+        });
+}
+
+function ensureElements (command) {
```

Comment:
Let's name it `ensureCommandElements` so it will not be confused with `ensureElement`